static/app/views/settings/projectMetrics/metricsExtractionRuleForm.tsx

```diff
@@ -46,21 +46,85 @@ interface Props extends Omit<FormProps, 'onSubmit'> {
 }
 
 const HIGH_CARDINALITY_TAGS = new Set([
+  SpanIndexedField.HTTP_RESPONSE_CONTENT_LENGTH,
   SpanIndexedField.SPAN_DURATION,
   SpanIndexedField.SPAN_SELF_TIME,
+  SpanIndexedField.SPAN_GROUP,
+  SpanIndexedField.ID,
+  SpanIndexedField.SPAN_AI_PIPELINE_GROUP,
+  SpanIndexedField.TRANSACTION_ID,
   SpanIndexedField.PROJECT_ID,
+  SpanIndexedField.PROFILE_ID,
+  SpanIndexedField.REPLAY_ID,
+  SpanIndexedField.TIMESTAMP,
+  SpanIndexedField.USER,
+  SpanIndexedField.USER_ID,
+  SpanIndexedField.USER_EMAIL,
+  SpanIndexedField.USER_USERNAME,
   SpanIndexedField.INP,
   SpanIndexedField.INP_SCORE,
   SpanIndexedField.INP_SCORE_WEIGHT,
   SpanIndexedField.TOTAL_SCORE,
   SpanIndexedField.CACHE_ITEM_SIZE,
+  SpanIndexedField.MESSAGING_MESSAGE_ID,
   SpanIndexedField.MESSAGING_MESSAGE_BODY_SIZE,
   SpanIndexedField.MESSAGING_MESSAGE_RECEIVE_LATENCY,
   SpanIndexedField.MESSAGING_MESSAGE_RETRY_COUNT,
-  SpanIndexedField.TRANSACTION_ID,
-  SpanIndexedField.ID,
 ]);
 
+const HIGH_CARDINALITY_KEYWORDS = new Set([
+  'id',
+  'slug',
+  'duration',
+  'time',
+  'timestamp',
+  'size',
+  'length',
+  'len',
+  'count',
+  'uuid',
+  'guid',
+]);
+
+/**
+ * Generates permutations of how a keyword can be formatted in a tag.
+ */
+const generateTagNamePermutation = (keyword: string): string[] => [
+  `_${keyword}`,
+  `${keyword}_`,
+  `.${keyword}`,
+  `${keyword}.`,
+  `-${keyword}`,
+  `${keyword}-`,
+  `:${keyword}`,
+  `${keyword}:`,
+  keyword.charAt(0).toUpperCase() + keyword.slice(1), // capitalize first letter
+  keyword.toUpperCase(), // uppercase version of tag
+];
+
+/**
+ * Checks if the tag is well known high cardinality tag or one of the known high cardinality keywords
+ * is in the tag. Permutations of a keyword are used as a heuristic to decrease the chance of
+ * false positives - e.g. 'idealTag' would not be considered high cardinality.
+ * NOTE: this is a heuristic and not a perfect solution - it can still result
+ * in false positives (e.g. 'myIdealTag' will be flagged as high cardinality tag)
```

Comment:
Let's err on the safe side here and avoid such false positives. While we want to prevent obvious errors, it's better to give the user more flexibility at this point. The cardinality limiter should still prevent too much damage.

Reply:
Removed from now, we will add more logic once we have data to query and to check cardinality